.pre-commit-config.yaml

```diff
@@ -29,7 +29,8 @@ repos:
 #    rev: v0.960
 #    hooks:
 #      - id: mypy
-#        files: "^pystac_client/.*\\.py$"
+#        files: ".*\\.py$"
+#        exclude: ^(docs/.*|tests/.*)$
```

Comment:
How close do you think we are to getting `tests` mypy happy?

Reply:
not too far -- I just wanted us to get these enabled for the regular code first before working on tests and docs

Reply:
I just ran it on tests and there's a probably 20-30 mypy warnings.